### PR TITLE
YARN-11561. [Federation] GPG Supports Format PolicyStateStore.

### PR DESCRIPTION
**Pull request by slfan1989**
### Description of PR

JIRA: YARN-11561. [Federation] GPG Supports Format PolicyStateStore.

### How was this patch tested?

Add Junit Test.

### For code changes:

- [ ] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

**Comment by slfan1989** [comment]
@goiri Can you help review this PR? Thank you very much!

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 57s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  buf  |   0m  0s |  |  buf was not available.  |
| +0 :ok: |  buf  |   0m  0s |  |  buf was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 27s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  31m 14s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 25s |  |  trunk passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |   2m 12s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   1m 20s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m  5s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   2m 15s |  |  trunk passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 52s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  spotbugs  |   3m 54s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  32m 13s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 32s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 39s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 16s |  |  the patch passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  cc  |   2m 16s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 16s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m  8s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  cc  |   2m  8s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m  8s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   1m  8s | [/results-checkstyle-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6300/1/artifact/out/results-checkstyle-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server.txt) |  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server: The patch generated 4 new + 1 unchanged - 0 fixed = 5 total (was 1)  |
| +1 :green_heart: |  mvnsite  |   1m 41s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 36s |  |  the patch passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 32s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  spotbugs  |   4m 10s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  32m 14s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 54s |  |  hadoop-yarn-server-common in the patch passed.  |
| +1 :green_heart: |  unit  | 106m 44s |  |  hadoop-yarn-server-resourcemanager in the patch passed.  |
| +1 :green_heart: |  unit  |   1m  0s |  |  hadoop-yarn-server-globalpolicygenerator in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 38s |  |  The patch does not generate ASF License warnings.  |
|  |   | 261m  1s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6300/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6300 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets cc buflint bufcompat |
| uname | Linux 26f87b44b77f 5.15.0-88-generic #98-Ubuntu SMP Mon Oct 2 15:18:56 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 2301a431ee17f6683c36d277713f1d16afc11a23 |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6300/1/testReport/ |
| Max. process+thread count | 948 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-globalpolicygenerator U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6300/1/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |  11m 11s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  buf  |   0m  0s |  |  buf was not available.  |
| +0 :ok: |  buf  |   0m  0s |  |  buf was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 28s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  31m 42s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 52s |  |  trunk passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |   2m 32s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   1m 28s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m  9s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   2m 15s |  |  trunk passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 48s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  spotbugs  |   3m 56s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  32m 45s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 31s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 34s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 17s |  |  the patch passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  cc  |   2m 17s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 17s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m  4s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  cc  |   2m  4s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m  4s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| -0 :warning: |  checkstyle  |   1m  9s | [/results-checkstyle-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server.txt](https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6300/2/artifact/out/results-checkstyle-hadoop-yarn-project_hadoop-yarn_hadoop-yarn-server.txt) |  hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server: The patch generated 4 new + 1 unchanged - 0 fixed = 5 total (was 1)  |
| +1 :green_heart: |  mvnsite  |   1m 43s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 33s |  |  the patch passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 29s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  spotbugs  |   4m  5s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  31m 48s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 26s |  |  hadoop-yarn-server-common in the patch passed.  |
| +1 :green_heart: |  unit  | 100m 15s |  |  hadoop-yarn-server-resourcemanager in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 52s |  |  hadoop-yarn-server-globalpolicygenerator in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 35s |  |  The patch does not generate ASF License warnings.  |
|  |   | 265m 21s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6300/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6300 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets cc buflint bufcompat |
| uname | Linux be3c6964df5d 5.15.0-88-generic #98-Ubuntu SMP Mon Oct 2 15:18:56 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / 2301a431ee17f6683c36d277713f1d16afc11a23 |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6300/2/testReport/ |
| Max. process+thread count | 988 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-globalpolicygenerator U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6300/2/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-globalpolicygenerator/src/main/java/org/apache/hadoop/yarn/server/globalpolicygenerator/GlobalPolicyGenerator.java [hunk 1]
```diff
@@ -366,4 +365,15 @@ public WebApp getWebApp() {
...
   }
+
+  private static void handFormatPolicyStateStore(Configuration conf) {
+    try {
+      System.out.println("Deleting Federation policy state store.");
+      FederationStateStoreFacade facade = FederationStateStoreFacade.getInstance(conf);
+      System.out.println("Federation policy state store has been cleaned.");
+      facade.deleteAllPoliciesConfigurations();
+    } catch (Exception e) {
+      System.err.println("Delete Federation policy state store error, exception = " + e);
```
Comment (goiri) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-globalpolicygenerator/src/main/java/org/apache/hadoop/yarn/server/globalpolicygenerator/GlobalPolicyGenerator.java:377]: Should this be LOG? Or this is for the user?

**Review by goiri — APPROVED** [review]

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-globalpolicygenerator/src/main/java/org/apache/hadoop/yarn/server/globalpolicygenerator/GlobalPolicyGenerator.java (hunk 1 above)
Comment (slfan1989) [reply hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-globalpolicygenerator/src/main/java/org/apache/hadoop/yarn/server/globalpolicygenerator/GlobalPolicyGenerator.java:377]: Thanks for your suggestion!  We should use `LOG`. I will improve this part of the code.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 32s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +0 :ok: |  buf  |   0m  0s |  |  buf was not available.  |
| +0 :ok: |  buf  |   0m  0s |  |  buf was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  |  The patch appears to include 1 new or modified test files.  |
|||| _ trunk Compile Tests _ |
| +0 :ok: |  mvndep  |  14m 33s |  |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |  30m 27s |  |  trunk passed  |
| +1 :green_heart: |  compile  |   2m 24s |  |  trunk passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  compile  |   2m 13s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  checkstyle  |   1m 17s |  |  trunk passed  |
| +1 :green_heart: |  mvnsite  |   2m  1s |  |  trunk passed  |
| +1 :green_heart: |  javadoc  |   1m 57s |  |  trunk passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 46s |  |  trunk passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  spotbugs  |   3m 52s |  |  trunk passed  |
| +1 :green_heart: |  shadedclient  |  31m 33s |  |  branch has no errors when building and testing our client artifacts.  |
|||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 31s |  |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   1m 33s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 17s |  |  the patch passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  cc  |   2m 17s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 17s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   2m  4s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  cc  |   2m  4s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   2m  4s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   1m  7s |  |  the patch passed  |
| +1 :green_heart: |  mvnsite  |   1m 41s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   1m 34s |  |  the patch passed with JDK Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04  |
| +1 :green_heart: |  javadoc  |   1m 29s |  |  the patch passed with JDK Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05  |
| +1 :green_heart: |  spotbugs  |   4m  9s |  |  the patch passed  |
| +1 :green_heart: |  shadedclient  |  31m 41s |  |  patch has no errors when building and testing our client artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   3m 26s |  |  hadoop-yarn-server-common in the patch passed.  |
| +1 :green_heart: |  unit  | 100m 43s |  |  hadoop-yarn-server-resourcemanager in the patch passed.  |
| +1 :green_heart: |  unit  |   0m 52s |  |  hadoop-yarn-server-globalpolicygenerator in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 35s |  |  The patch does not generate ASF License warnings.  |
|  |   | 250m 58s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6300/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/6300 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient spotbugs checkstyle codespell detsecrets cc buflint bufcompat |
| uname | Linux e0354c9c5ece 5.15.0-88-generic #98-Ubuntu SMP Mon Oct 2 15:18:56 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/bin/hadoop.sh |
| git revision | trunk / afd78296c30c2bfa23a73f18abcd77e5f081a35a |
| Default Java | Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
| Multi-JDK versions | /usr/lib/jvm/java-11-openjdk-amd64:Ubuntu-11.0.20.1+1-post-Ubuntu-0ubuntu120.04 /usr/lib/jvm/java-8-openjdk-amd64:Private Build-1.8.0_382-8u382-ga-1~20.04.1-b05 |
|  Test Results | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6300/3/testReport/ |
| Max. process+thread count | 967 (vs. ulimit of 5500) |
| modules | C: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-globalpolicygenerator U: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server |
| Console output | https://ci-hadoop.apache.org/job/hadoop-multibranch/job/PR-6300/3/console |
| versions | git=2.25.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.14.0 https://yetus.apache.org |


This message was automatically generated.

**Review by goiri — APPROVED** [review]

**Comment by slfan1989** [comment]
@goiri Thank you very much for your help in reviewing the code!

**Pull request merged**